client/windows/CMessage.cpp

```diff
@@ -60,125 +60,222 @@ void CMessage::dispose()
 		item.reset();
 }
 
-std::vector<std::string> CMessage::breakText(std::string text, size_t maxLineWidth, EFonts font)
+bool CMessage::validateTags(
+	const std::vector<std::string_view::const_iterator> & openingTags,
+	const std::vector<std::string_view::const_iterator> & closingTags)
```

Comment:
Not sure about using string view to be honest - it is too fragile, even more fragile than pointers.
Example:
```cpp
void foo(std::string_view value);

int main()
{
    std::string someString = "helloworld";
    foo(std::string("abcd")); // Illegal! std::string is destroyed after constructing its string view
    foo(someString.substr(1,2)); // Illegal! substr returns temporary string that will be destroyed
}
```

Your code works only because `std::string_view::const_iterator` is basically `const char*`, but this is clearly undefined behavior.

IMO we should use std::strings (they usually have short string optimization anyway) and only consider other approaches for performance-critical parts.

Reply:
> IMO we should use std::strings (they usually have short string optimization anyway) and only consider other approaches for performance-critical parts.

Ok, I'll try to do it based on string